source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -193,21 +207,28 @@ void HeaderToMetadataFilter::setEncoderFilterCallbacks(
 
 bool HeaderToMetadataFilter::addMetadata(StructMap& struct_map, const std::string& meta_namespace,
                                          const std::string& key, std::string value, ValueType type,
-                                         ValueEncode encode) const {
+                                         ValueEncode encode, HeaderDirection direction) const {
   ProtobufWkt::Value val;
+  const auto* config = getConfig();
 
   ASSERT(!value.empty());
 
   if (value.size() >= MAX_HEADER_VALUE_LEN) {
     // Too long, go away.
     ENVOY_LOG(debug, "metadata value is too long");
+    if (config->stats().has_value()) {
+      config->stats().value().header_value_too_long_.inc();
+    }
     return false;
   }
 
   if (encode == envoy::extensions::filters::http::header_to_metadata::v3::Config::BASE64) {
     value = Base64::decodeWithoutPadding(value);
     if (value.empty()) {
       ENVOY_LOG(debug, "Base64 decode failed");
+      if (config->stats().has_value()) {
+        config->stats().value().base64_decode_failed_.inc();
+      }
```

Comment:
I will inclined to createa method in the `Config` to accept the two enums, one enum for event, one enum for traffic direction.
Then handle all this logic in same method to avoid this `if (config->stats().has_value())` everywhere. If we need to add new stats in the future, it would much simpler also.